awscli/customizations/s3/subcommands.py

```diff
@@ -1359,6 +1382,19 @@ def _raise_if_mv_same_paths(self, src, dest):
                 f"{self.parameters['src']} - {self.parameters['dest']}"
             )
 
+    def _raise_if_paths_type_incorrect_for_param(self, param, paths_type, expected_paths_type):
+        if paths_type != expected_paths_type:
+            expected_usage_map = {
+                'locals3': '<LocalPath> <S3Uri>',
+                's3s3': '<S3Uri> <S3Uri>',
+                's3local': '<S3Uri> <LocalPath>',
+                's3': '<S3Uri>'
+            }
+            raise ParamValidationError(
+                "The %s flag is only compatible with the format: %s. You used the format: %s." %
+                (param, expected_usage_map[expected_paths_type], expected_usage_map[paths_type])
```

Comment:
```suggestion
                f"Expected {param} parameter to be used with one of following path formats: "
                f"{', '.join(allowed_paths)} but received {expected_usage_map[paths_type]}"
```
Where `allowed_paths` is a list of the mapped paths types.